packages/core/lib/segments/segment.js

```diff
@@ -346,18 +338,26 @@ Segment.prototype.close = function(err, remote) {
 Segment.prototype.flush = function flush() {
   if (this.notTraced !== true) {
     delete this.exception;
-    SegmentEmitter.send(omit(this, ['counter', 'notTraced']));
+
+    var thisCopy = Object.assign({}, this);
+    if ('counter' in thisCopy) delete thisCopy.counter;
+    if ('notTraced' in thisCopy) delete thisCopy.notTraced;
+
+    SegmentEmitter.send(thisCopy);
   }
 };
 
 Segment.prototype.format = function format() {
-  var ignore = ['counter', 'notTraced', 'exception'];
+  var thisCopy = Object.assign({}, this);
+
+  if ('counter' in thisCopy) delete thisCopy.counter;
+  if ('notTraced' in thisCopy) delete thisCopy.notTraced;
+  if ('exception' in thisCopy) delete thisCopy.exception;
 
-  if (isEmpty(this.subsegments))
-    ignore.push('subsegments');
+  if (Array.isArray(this.subsegments) && !this.subsegments.length && 'subsegments' in thisCopy)
+    delete thisCopy.subsegments;
```

Comment:
Nit: Same as above. Since the body of the if statement is on a separate line, can you wrap this in brackets? That'll make it more clear what our intent is.